docker/jax.dockerfile

```
@@ -0,0 +1,45 @@
+
+ARG BASE_IMAGE=ubuntu:18.04
+ARG PYTHON_VERSION=3.7
+ARG JAX_VERSION=
+# Setup develop base image packages(build-essentials etc)
+FROM ${BASE_IMAGE} as dev-base
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        ccache \
+        cmake \
+        curl \
+        git \
+        libjpeg-dev \
+        libpng-dev && \
+    rm -rf /var/lib/apt/lists/*
+RUN /usr/sbin/update-ccache-symlinks
+RUN mkdir /opt/ccache && ccache --set-config=cache_dir=/opt/ccache
+ENV PATH /opt/conda/bin:$PATH
+
+# Setup Miniconda
+FROM dev-base as conda
+ARG PYTHON_VERSION=3.7
+RUN curl -fsSL -v -o ~/miniconda.sh -O  https://repo.anaconda.com/miniconda/Miniconda3-py39_4.9.2-Linux-x86_64.sh  && \
```

```suggestion
RUN curl -fsSL -v -o ~/miniconda.sh -O  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
```
It may be better to run with the latest miniconda version rather than a given version string.

---

Just wanted to know if it's good idea to go with Conda, conda builds takes lot of space like 1.6 GB but on the other hand builds using default Python is around 600MB which is huge space-save. I have added new Dockerfiles with some more robust method than last one and pushed all builds using Conda into separate folder.     

Will it be good idea to keep Conda builds separate?

---

Good question, and great job on the changes! I think for now, keeping them separate is a good idea. Given the large difference in size, I think I am in favour of the smaller images, though we can keep both for the moment and see are there situations where Conda makes more sense.